analysis/src/TypeUtils.ml

```diff
@@ -34,8 +40,9 @@ let findTypeViaLoc ~full ~debug (loc : Location.t) =
   | Some {locType = Typed (_, typExpr, _)} -> Some typExpr
   | _ -> None
 
-let pathFromTypeExpr (t : Types.type_expr) =
+let rec pathFromTypeExpr (t : Types.type_expr) =
   match t.desc with
+  | Tconstr (Pident {name = "function$"}, [t; _], _) -> pathFromTypeExpr t
```

Comment:
Can revert, irrelevant in this repo. I suspect that makes the `rec` revertable too.